### PR TITLE
:sparkles: Add assessed archetypes section in drawer

### PR DESCRIPTION
**Pull request by ibolton336**
Resolves: 
https://issues.redhat.com/browse/MTA-1834
[MTA-1878](https://issues.redhat.com/browse/MTA-1878)
[Inconsistent hover text message for Assessment and Review columns](https://issues.redhat.com/browse/MTA-1835)

Screenshot 2023-12-13 at 12 15 39 PM
Screenshot 2023-12-13 at 12 15 32 PM

### CODE REVIEW

**Review by sjd78 — APPROVED** [review]
Behavior for application with an archetype but the archetype is missing an assessment or review is looking good now.  The "None" shows as expected.

File: client/src/app/components/IconedStatus.tsx [hunk 1]
```diff
@@ -89,16 +111,27 @@ export const IconedStatus: React.FC<IIconedStatusProps> = ({
     },
   };
   const presetProps = preset && presets[preset];
+  console.log("presetProps", presetProps);
```
Comment (sjd78) [inline client/src/app/components/IconedStatus.tsx:114]: Log!

File: client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx [hunk 2]
```diff
@@ -1,50 +1,64 @@
...
+      (assessment) =>
+        assessment.status === "started" || assessment.status === "complete"
+    )
   ) {
-    return <IconedStatus preset="InProgress" />;
+    statusPreset = "InProgress";
   }
-
-  return <IconedStatus preset="NotStarted" />;
+  console.log("tooltipCount", tooltipCount);
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx:62]: Log!

**Review by sjd78 — CHANGES_REQUESTED** [review]
I'm not getting the inherited text where I saw the infotip icon before.

File: client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx [hunk 3]
```diff
@@ -1,50 +1,64 @@
...
   const { t } = useTranslation();
 
+  const { archetypes, isFetching } = useFetchArchetypes();
+
+  const applicationArchetypes = application.archetypes?.map((archetypeRef) => {
+    return archetypes?.find((archetype) => archetype.id === archetypeRef.id);
+  });
+
+  const hasAssessedArchetype = applicationArchetypes?.some(
+    (archetype) => !!archetype?.assessments?.length
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx:26]: Nit: I think I prefer the actual `> 0` check here, but as-is is ok.

File: client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx [hunk 4]
```diff
@@ -1,50 +1,64 @@
...
+  if (isFetching || isFetchingAssessmentsById) {
     return <Spinner size="md" />;
   }
 
-  if (
-    assessments?.some((a) => a.status === "started" || a.status === "complete")
+  let statusPreset: IconedStatusPreset = "NotStarted"; // Default status
+  let tooltipCount: number = 0;
+
+  if (application?.assessed) {
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx:46]: Looks like this flag is set `true` if the application is directly assessed OR if it is inheriting an archetype assessment.

Suggestion:
```tsx
  const isDirectlyAssessed =
    application.assessed && (application.assessments?.length ?? 0) > 0;

  if (isDirectlyAssessed) {
    ...
```

Comment (ibolton336) [reply client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx:46]: @sjd78 The application.assessed bool is reflecting the app's archetype assessed state, also.

Comment (ibolton336) [reply client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx:46]: Ah I think I follow you.

File: client/src/app/pages/applications/components/application-review-status/application-review-status.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  isLoading?: boolean;
+}
+
+export const ApplicationReviewStatus: React.FC<
+  ApplicationReviewStatusProps
+> = ({ application }) => {
+  const { t } = useTranslation();
+
+  const { archetypes, isFetching } = useFetchArchetypes();
+  const isAppReviewed = !!application.review;
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-review-status/application-review-status.tsx:20]: Is the review one for the application or does it include the archetype review?  This column may have the same issue as the assessment column...

**Review by sjd78 — COMMENTED** [review]
Everything is looking good, just a question on the review column.

File: client/src/app/pages/applications/components/application-review-status/application-review-status.tsx (hunk 5 above)
Comment (ibolton336) [reply client/src/app/pages/applications/components/application-review-status/application-review-status.tsx:20]: The review only includes the app level review. There is no review bool, this is just the actual review object on the application.

**Review by sjd78 — APPROVED** [review]
LGTM

**Pull request merged**